components/treelist/toolbar.md

```diff
@@ -10,145 +10,227 @@ position: 45
-            Employee root = new Employee
+            if (item.Equals(items[i]))
             {
-                Id = LastId,
-                Name = $"root: {i}",
-                EmailAddress = $"{i}@example.com",
-                HireDate = DateTime.Now.AddYears(-i),
-                DirectReports = new List<Employee>(),
-                HasChildren = true
-            };
-            data.Add(root);
-            LastId++;
+                items.Remove(item);
 
-            for (int j = 1; j < 4; j++)
+                return;
+            }
+            else if (items[i].Items?.Count > 0)
             {
-                int currId = LastId;
-                Employee firstLevelChild = new Employee
-                {
-                    Id = currId,
-                    Name = $"first level child {j} of {i}",
-                    EmailAddress = $"{currId}@example.com",
-                    HireDate = DateTime.Now.AddDays(-currId),
-                    DirectReports = new List<Employee>(),
-                    HasChildren = true
-                };
-                root.DirectReports.Add(firstLevelChild);
-                LastId++;
+                RemoveChildRecursive(items[i].Items, item);
 
-                for (int k = 1; k < 3; k++)
+                if (items[i].Items.Count == 0)
                 {
-                    int nestedId = LastId;
-                    firstLevelChild.DirectReports.Add(new Employee
-                    {
-                        Id = LastId,
-                        Name = $"second level child {k} of {j} and {i}",
-                        EmailAddress = $"{nestedId}@example.com",
-                        HireDate = DateTime.Now.AddMinutes(-nestedId)
-                    }); ;
-                    LastId++;
+                    items[i].HasChildren = false;
                 }
             }
         }
+    }
 
-        return await Task.FromResult(data);
+    private void CreateItem(TreeListCommandEventArgs args)
+    {
+        var argsItem = args.Item as HierarchicalModel;
+
+        argsItem.Id = LastId++;
+
+        if (args.ParentItem != null)
+        {
+            var parent = (HierarchicalModel)args.ParentItem;
+
+            parent.HasChildren = true;
+            if (parent.Items == null)
+            {
+                parent.Items = new List<HierarchicalModel>();
+            }
+
+            parent.Items.Insert(0, argsItem);
+        }
+        else
+        {
+            Data.Insert(0, argsItem);
+        }
     }
-}
 
+    public class HierarchicalModel
+    {
+        public int Id { get; set; }
+        public string FirstName { get; set; }
+        public DateTime HireDate { get; set; }
+        public List<HierarchicalModel> Items { get; set; }
+        public bool HasChildren { get; set; }
+        public bool OutOfOffice { get; set; }
+    }
+}
 ````
 
->caption The result from the code snippet above, after the custom command button in the toolbar was clicked
+## Custom Toolbar Configuration
```

Comment:
I have some concerns about this heading, because the word "custom" doesn't convey any meaning. Each toolbar configuration is unique, so it's not clear what is the "custom" part. If you don't like "Toobar Template" more, you can leave it as is.